src/test/java/com/cloudant/tests/util/Utils.java

```diff
@@ -66,13 +73,27 @@ public static String getHostName(String account) {
     public static void removeReplicatorTestDoc(CloudantClient account, String replicatorDocId)
             throws Exception {
 
-        //Grab replicator doc revision using HTTP GET command
+        //Grab replicator doc revision using HTTP HEAD command
         String replicatorDb = "_replicator";
         URI uri = URI.create(account.getBaseUri() + replicatorDb + "/"
                 + replicatorDocId);
         HttpConnection head = Http.HEAD(uri);
-        String revision = head.execute().getConnection().getHeaderField("ETag");
 
+        //add a response interceptor to allow us to retrieve the ETag revision header
+        final AtomicReference<String> revisionRef = new AtomicReference<String>();
```

Comment:
Honestly this was just for conciseness to get a getter/setter for free on a final field so the `HttpConnectionResponseInterceptor` instance used here could remain anonymous.
The alternative is the significantly more verbose declaration of an inner (local) class that is a named subclass of `HttpConnectionResponseInterceptor` that includes a getter method to return the `revision` string, which I'm willing to do it if it is preferred or happy to receive suggestions for other alternatives.